validator/client/beacon-api/propose_beacon_block.go

```diff
@@ -62,7 +64,42 @@ func (c *beaconApiValidatorClient) proposeBeaconBlock(ctx context.Context, in *e
 	}
 
 	headers := map[string]string{"Eth-Consensus-Version": res.consensusVersion}
-	err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(res.marshalledJSON), nil)
+
+	// Try PostSSZ first with SSZ data
+	if res.marshalledSSZ != nil {
+		headers["Content-Type"] = "application/octet-stream"
+		_, _, err = c.jsonRestHandler.PostSSZ(ctx, endpoint, headers, bytes.NewBuffer(res.marshalledSSZ))
+		if err != nil {
+			errJson := &httputil.DefaultJsonError{}
+			// If PostSSZ fails with 406 (Not Acceptable), fall back to JSON
+			if errors.As(err, &errJson) && errJson.Code == http.StatusNotAcceptable && res.marshalJSON != nil {
+				log.WithError(err).Warn("Failed to submit block ssz, falling back to JSON")
+				jsonData, jsonErr := res.marshalJSON()
+				if jsonErr != nil {
+					return nil, errors.Wrap(jsonErr, "failed to marshal JSON fallback")
+				}
+				// Reset headers for JSON
+				headers = map[string]string{"Eth-Consensus-Version": res.consensusVersion}
+				err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(jsonData), nil)
+				// If JSON also fails, return that error; otherwise continue
```

Comment:
```suggestion
				// If JSON also fails, return that error
```
I am not sure what `otherwise continue` refers to. This `if` doesn't have an `else`, so maybe we can just remove this part?